packages/tizen_bundle/lib/tizen_bundle.dart

```diff
@@ -0,0 +1,277 @@
+  bool _isDisposed = false;
+  static final Finalizer<Bundle> _finalizer =
+      Finalizer<Bundle>((Bundle bundle) => bundle.dispose());
+
+  /// The keys of this.
+  @override
+  Iterable<String> get keys => toMap().keys;
+
+  /// The value for the given key, or null if key is not in the Bundle.
+  @override
+  Object? operator [](Object? key) {
+    if (key == null) {
+      return null;
+    }
+
+    Object? value;
+    final int type = _getType(key as String);
+    if (type == bundle_type.BUNDLE_TYPE_STR) {
+      value = _getString(key);
+    } else if (type == bundle_type.BUNDLE_TYPE_STR_ARRAY) {
+      value = _getStrings(key);
+    } else if (type == bundle_type.BUNDLE_TYPE_BYTE) {
+      value = _getBytes(key);
+    }
+
+    return value;
+  }
+
+  /// Associates the key with the given value.
+  @override
+  void operator []=(String key, Object value) {
+    if (this[key] != null) {
+      remove(key);
+    }
+
+    if (value is String) {
+      _addString(key, value);
+    } else if (value is List<String>) {
+      _addStrings(key, value);
+    } else if (value is Uint8List) {
+      _addBytes(key, value);
+    } else {
+      _throwException(bundle_error_e.BUNDLE_ERROR_INVALID_PARAMETER);
+    }
+  }
+
+  /// Removes all entries from the [Bundle].
+  @override
+  void clear() {
+    keys.forEach(remove);
+  }
+
+  static void _bundleIteratorCallback(Pointer<Int8> pKey, int type,
+      Pointer<keyval_t> pKeyval, Pointer<Void> userData) {
+    _keys.add(pKey.toDartString());
+  }
+
+  /// Creates a [Map] containing the elements of this [Bundle].
+  Map<String, Object> toMap() {
+    _keys.clear();
+    tizen.bundle_foreach(
+        _handle, Pointer.fromFunction(_bundleIteratorCallback), nullptr);
+    final Map<String, Object> map = <String, Object>{};
+    for (final String element in _keys) {
+      map[element] = this[element]!;
+    }
+    return map;
+  }
+
+  /// Removes [key] and its associated value, if present, from the [Bundle].
+  @override
+  void remove(Object? key) {
+    if (key == null) {
+      return;
+    }
+
+    final int ret = using((Arena arena) {
+      final String keyName = key as String;
+      return tizen.bundle_del(_handle, keyName.toNativeInt8(allocator: arena));
+    });
+    if (ret != bundle_error_e.BUNDLE_ERROR_NONE) {
+      _throwException(ret);
+    }
+  }
+
+  /// Converts this object to String type.
+  String toRaw() {
+    final String raw = using((Arena arena) {
+      final Pointer<Pointer<Int8>> rawPointer = arena<Pointer<Int8>>();
+      final Pointer<Int32> lengthPointer = arena<Int32>();
+      final int ret = tizen.bundle_encode(
+          _handle, rawPointer.cast<Pointer<Uint8>>(), lengthPointer);
+      if (ret != bundle_error_e.BUNDLE_ERROR_NONE) {
+        _throwException(ret);
+      }
+
+      return rawPointer.value.toDartString();
+    });
+
+    return raw;
+  }
+
+  /// Releases all resources associated with this object.
+  void dispose() {
```

Comment:
It's the intentional code.
The user(developer) can release it when it is no longer needed.

Reply:
Unlike C#, Dart doesn't have built-in support for `using` and the IDisposable pattern so it's up to the API designer's choice. If we don't provide the `dispose` method, the object will automatically be disposed by the garbage collector so that the user doesn't have to care about the object's lifecycle (I personally prefer this). If there's a specific case where the user has to release the object at specific point of time, we need to provide the `dispose` method or a similar mechanism.

Note: tizen_app_manager's [`AppRunningContext.dispose`](https://github.com/flutter-tizen/plugins/blob/ccd824c1a20acd19285d0492efbd1407015e06f4/packages/tizen_app_manager/lib/app_manager.dart#L200) has been deprecated and will be removed in the future.

Reply:
Thank you for explanation.
The dispose method is removed.